### DISCUSSION TITLE
Please state the difference between fetch-depth and fetch_depth

### DISCUSSION THREAD

**Discussion by jackton1** [tj-actions/changed-files#1411] [Q&A] [answered]
What is the difference between fetch-depth and fetch_depth? fetch-depth is mentioned here: IMPORTANT: For push events when configuring [actions/checkout](https://github.com/actions/checkout#usage) the fetch-depth should be set to either fetch-depth: 0 OR fetch-depth: 2 depending on your use case; and is in the examples. However when used, an unrecognized error is given and the you are informed to use fetch_depth. fetch_depth is in the Input Table and is defaulted to 50.

So which one are you supposed to use?

Reference Issue: https://github.com/tj-actions/changed-files/issues/1410

**Comment by jackton1** [comment, marked as answer]
The distinction between the `fetch-depth` passed to the [checkout action](https://github.com/actions/checkout) and the `fetch_depth` used by [tj-actions/changed-files](https://github.com/tj-actions/changed-files) is as follows:

### [checkout action](https://github.com/actions/checkout)
This action checks-out your repository under `$GITHUB_WORKSPACE`, so the [tj-actions/changed-files](https://github.com/tj-actions/changed-files) workflow can access it.

- `fetch-depth`: Indicates the number of commits to fetch.
  - 0: Represents all history for all branches and tags.
  - 1: Represents the current commit `HEAD`
  - 2: Represents the current commit and its parent `HEAD^1`

### [tj-actions/changed-files](https://github.com/tj-actions/changed-files)
Uses either the local git repository or Github's REST API to check for changes that have occurred to files.

- `fetch_depth`: Controls the number of additional commits that should be pulled when a shallow clone of the repository is performed by the [checkout action](https://github.com/actions/checkout).

## Questions
- Q: I already pulled all the necessary history by setting `fetch-depth` to `0`. Do I still need to pass additional depth information to [tj-actions/changed-files](https://github.com/tj-actions/changed-files)?
> **No, since you already have all history locally, no additional history would be pulled.**

- Q: When would I need to set [tj-actions/changed-files](https://github.com/tj-actions/changed-files) `fetch_depth` input?
> **When you performed a shallow clone and still want to control the number of additional commits that [tj-actions/changed-files](https://github.com/tj-actions/changed-files) should pull.**

- Q: Can I prevent additional history from getting pulled for a shallow clone of the repository?
> **Yes, you can set `skip_initial_fetch` to `true`.**

**Reply by rick-kenna** [reply]
@jackton1 Thank-you for an understandable and quick response.

**Reply by jackton1** [reply]
Anytime
